btclightclient/test_utils.go

```diff
@@ -0,0 +1,117 @@
+package btclightclient
+
+import (
+	"crypto/sha256"
+	"strconv"
+	"strings"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+func GenerateBlock(
+	prevHash chainhash.Hash,
+	miner string,
+	transactions []*btcutil.Tx,
+	difficulty int,
+) (*wire.MsgBlock, error) {
+	block := &wire.MsgBlock{}
+	targetPrefix := strings.Repeat("0", difficulty)
+
+	for nonce := 0; ; nonce++ {
+		txnStrings := ""
+		var txHashes []*chainhash.Hash
+		for _, tx := range transactions {
+			txHash := tx.Hash()
+			txnStrings += txHash.String()
+			txHashes = append(txHashes, txHash)
+		}
+		hashInput := prevHash.String() + txnStrings + miner + strconv.Itoa(nonce)
+		hash := sha256.Sum256([]byte(hashInput))
+		var blockHash chainhash.Hash
+		copy(blockHash[:], hash[:])
+
+		if strings.HasPrefix(blockHash.String(), targetPrefix) {
+			block.Header.PrevBlock = prevHash
+			block.Header.Nonce = uint32(nonce)
+			merkleRoot := CalculateMerkleRoot(txHashes)
+			block.Header.MerkleRoot = *merkleRoot[len(merkleRoot)-1]
+			for _, tx := range transactions {
+				block.Transactions = append(block.Transactions, tx.MsgTx())
+			}
+			return block, nil
+		}
+	}
+}
+
+// CalculateMerkleRoot creates a merkle tree from a slice of transaction hashes.
+func CalculateMerkleRoot(transactions []*chainhash.Hash) []*chainhash.Hash {
```

Comment:
btcd support https://github.com/btcsuite/btcd/blob/master/blockchain/merkle.go#L185
Can we ues this API?